components/gitpod-protocol/src/protocol.ts

```diff
@@ -95,6 +95,47 @@ export namespace User {
         }
         return undefined;
     }
+
+    export function hasPreferredIde(user: User) {
+        return (
+            typeof user?.additionalData?.ideSettings?.defaultIde !== "undefined" ||
+            typeof user?.additionalData?.ideSettings?.useLatestVersion !== "undefined"
+        );
+    }
+
+    export function isOnboardingUser(user: User) {
+        return !hasPreferredIde(user);
+    }
+
+    export function migrationIDESettings(user: User) {
```

Comment:
@mustard-mh @andreafalzetti Why did we pull it to protocol from dashboard?

Reply:
@akosyakov This? https://github.com/gitpod-io/gitpod/pull/9432#discussion_r858918692

Reply:
reads like you agree to put it in dashboard though, not in protocol

Reply:
we agreed to put it in protocol